### PR TITLE
add filter to customize existing progress bar to add `%` to it #1026

### PR DESCRIPTION
**Pull request by thrijith**
add a filter to customize existing progress bar to add `%` to it #1026

### CODE REVIEW

File: app/assets/js/rtMedia.backbone.js [hunk 1]
```diff
@@ -738,7 +738,15 @@ jQuery( function( $ ) {
 			uploaderObj.uploader.bind( 'UploadProgress', function( up, file ) {
 				//$("#" + file.id + " .plupload_file_status").html(file.percent + "%");
 				//$( "#" + file.id + " .plupload_file_status" ).html( rtmedia_uploading_msg + '( ' + file.percent + '% )' );
-				$( '#' + file.id + ' .plupload_file_status' ).html( '<div class="plupload_file_progress ui-widget-header" style="width: ' + file.percent + '%;"></div>' );
+				// creates a progress bar to display file upload status
+				var progressBar = jQuery( '<div/>', {
+					'class': 'plupload_file_progress ui-widget-header',
+				});
+				progressBar.css( 'width', file.percent + '%');
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: `progressBar.css( 'width', file.percent + '%' );`

Please make sure you follow coding standards.

File: app/assets/js/rtMedia.backbone.js [hunk 2]
```diff
@@ -1239,7 +1247,15 @@ jQuery( document ).ready( function( $ ) {
 
 		objUploadView.uploader.bind( 'UploadProgress', function( up, file ) {
 			//$( "#" + file.id + " .plupload_file_status" ).html( rtmedia_uploading_msg + '( ' + file.percent + '% )' );
-			$( '#' + file.id + ' .plupload_file_status' ).html( '<div class="plupload_file_progress ui-widget-header" style="width: ' + file.percent + '%;"></div>' );
+			// creates a progress bar to display file upload status
+			var progressBar = jQuery( '<div/>', {
+				'class': 'plupload_file_progress ui-widget-header',
+			});
+			progressBar.css( 'width', file.percent + '%');
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: `progressBar.css( 'width', file.percent + '%' );`

File: app/assets/js/rtMedia.backbone.js [hunk 3]
```diff
@@ -2431,7 +2447,15 @@ function renderUploadercomment_media( widget_id, parent_id_type ) {
 		} );
 
         commentObj[ widget_id ].uploader.bind( 'UploadProgress', function( up, file ) {
-			jQuery( '#' + file.id + ' .plupload_file_status' ).html( '<div class="plupload_file_progress ui-widget-header" style="width: ' + file.percent + '%;"></div>' );
+			// creates a progress bar to display file upload status
+			var progressBar = jQuery( '<div/>', {
+				'class': 'plupload_file_progress ui-widget-header',
+			});
+			progressBar.css( 'width', file.percent + '%');
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: `progressBar.css( 'width', file.percent + '%' );`

**Comment by kjnanda** [comment]
@thrijith I checked this at my end. 

- The progress bar shows directly shows `100%` for almost all the media. Which should not happen.
Please check this video here --> https://goo.gl/EX2rCZ

- The blank `progress bar` does not look good.

screen_shot_2018-01-02_at_11_13_53_am_jpg__1280x800_

**Comment by thrijith** [comment]
to get this filter working add [this ](https://gist.github.com/thrijith/6e452a38c101e611a11f8a932c15f84f) code in your theme's **function.php**

**Comment by kjnanda** [comment]
@thrijith With ref to this comment - https://github.com/rtMediaWP/rtMedia/pull/1031#issuecomment-354706489

This looks good on info site. For local site as I mentioned earlier I can not see progress bar.

**Comment by kjnanda** [comment]
@thrijith Please check this points as mentioned below.

- Sometimes the progress bar repeats after some count like shown in video. 

> For the first video - progress bar goes till `23` and then it again starts from `1` 
> It also repeats again after it reached to 100% while second video is being uploaded

-  It does not show `%` sign up to some count

Please check this link - https://docs.google.com/file/d/1mNMj_bmvMPxeNKzaplVB6QsQzPEX5yRy/edit

NOTE : This issue is specifically found in `2016` theme.

**Comment by thrijith** [comment]
reply to [this](https://github.com/rtMediaWP/rtMedia/pull/1031#issuecomment-354709738)

update [gitst](https://gist.github.com/thrijith/6e452a38c101e611a11f8a932c15f84f) to solve
- Sometimes the progress bar repeats after some count like shown in the video.

> For the first video - progress bar goes till 23 and then it again starts from 1
It also repeats again after it reached to 100% while second video is being uploaded

**Pull request merged**